crates/matrix-sdk-sqlite/src/event_cache_store.rs

```diff
@@ -627,6 +627,13 @@ impl EventCacheStore for SqliteEventCacheStore {
         room_id: &RoomId,
         events: Vec<OwnedEventId>,
     ) -> Result<Vec<OwnedEventId>, Self::Error> {
+        // If there's no events for which we want to check duplicates, we can return
+        // early. It's not only an optimization to do so: it's required, otherwise the
+        // `repeat_vars` call below will panic.
+        if events.is_empty() {
+            return Ok(Vec::new());
+        }
```

Comment:
Don't we have a similar problem in the other direction? When we have too many events and the number of parameters doesn't fit into a single query.

Reply:
No because I'm using `chunk_large_query_over` already.

Reply:
Ah Indeed.